Update destroy_by documentation not returning collection after Rails 7.1 [ci skip]

abhaynikam opened the pull request:
`destroy_by` internally uses `destroy_all`. In #40445
`destroy_all` collection return was changed if 
`config.active_record.destroy_all_in_batches` was set to true.

This PR update the documentation to highlight the changes 
and deprecation messages that would be shown.

abhaynikam committed:
Update destroy_by documentation not returning collection after Rails 7.1

`destroy_by` internally uses `destroy_all`. In #40445
`destroy_all` collection return was changed if `config.active_record.destroy_all_in_batches`
was set to true.

This PR update the documentation to highlight the changes.

p8 commented on activerecord/lib/active_record/relation.rb:
```diff
@@ -577,7 +577,7 @@ def touch_all(*names, time: nil)
     # If +config.active_record.destroy_all_in_batches+ is set to +true+, it will ensure
     # to perform the record's deletion in batches
-    # and destroy_all won't longer return the collection of the deleted records
+    # and +destroy_all+ will no longer return the collection of the deleted records
```
```suggestion
    # and +destroy_all+ will no longer return the collection of deleted records
```

p8 commented on activerecord/lib/active_record/relation.rb:
```diff
@@ -647,7 +647,12 @@ def delete_all
     # Finds and destroys all records matching the specified conditions.
     # This is short-hand for <tt>relation.where(condition).destroy_all</tt>.
-    # Returns the collection of objects that were destroyed.
+    # Returns the collection of objects that were destroyed if
```
```suggestion
    # Returns the collection of records that were destroyed if
```

p8 commented on activerecord/lib/active_record/relation.rb:
```diff
@@ -647,7 +647,12 @@ def delete_all
+    # +config.active_record.destroy_all_in_batches+ is set to +false+.
+    #
+    # If +config.active_record.destroy_all_in_batches+ is set to +true+, it will ensure
+    # to perform the record's deletion in batches
```
I'm reading this as meaning to be plural, but not sure 😄 
```suggestion
    # to perform the records deletion in batches
```

p8 commented on activerecord/lib/active_record/relation.rb:
```diff
@@ -647,7 +647,12 @@ def delete_all
+    #
+    # If +config.active_record.destroy_all_in_batches+ is set to +true+, it will ensure
+    # to perform the record's deletion in batches
+    # and +destroy_by+ will no longer return the collection of the deleted records
```
```suggestion
    # and +destroy_by+ will no longer return the collection of the destroyed records
```

intrip commented:
@abhaynikam please remove the [ci skip], we shouldn't skip anymore CI for docs: https://github.com/rails/rails/commit/6faacea22232c7d944e893c1948f707fa047cf21

abhaynikam commented:
@intrip Thanks for pointing that out. Removed CI skip from the commit messages. 😊 

@p8 Addressed all the comments. Can you please take another look?

p8 commented on activerecord/lib/active_record/relation.rb:
```diff
@@ -576,8 +576,8 @@ def touch_all(*names, time: nil)
-    # to perform the record's deletion in batches
-    # and destroy_all won't longer return the collection of the deleted records
+    # to perform the deletion of records in batches
+    # and +destroy_all+ will no longer return the collection of the destroyed records
```
```suggestion
    # and +destroy_all+ will no longer return the collection of the destroyed records.
```

p8 commented on activerecord/lib/active_record/relation.rb:
```diff
@@ -647,7 +647,12 @@ def delete_all
+    #
+    # If +config.active_record.destroy_all_in_batches+ is set to +true+, it will ensure
+    # to perform the deletion of records in batches
+    # and +destroy_by+ will no longer return the collection of the destroyed records
```
```suggestion
    # and +destroy_by+ will no longer return the collection of the destroyed records.
```

p8 commented on activerecord/lib/active_record/relation.rb:
```diff
@@ -576,8 +576,8 @@ def touch_all(*names, time: nil)
-    # to perform the record's deletion in batches
-    # and destroy_all won't longer return the collection of the deleted records
+    # to perform the deletion of records in batches
+    # and +destroy_all+ will no longer return the collection of the destroyed records.
```
Sorry, last one :)
```suggestion
    # and +destroy_all+ will no longer return the collection of destroyed records.
```

p8 commented:
Thanks @abhaynikam !

abhaynikam commented:
Closing this PR since changes were revered in #42763.
